kombu/transport/SQS.py

```diff
@@ -170,12 +181,21 @@ def _new_queue(self, queue, **kwargs):
         # _queue_cache population.
         queue = self.canonical_queue_name(queue)
 
+        # If we have the URL cached, return it
+        rv = self._queue_cache.get(queue, None)
+        if rv:
+            return rv
+
+        # If we don't have it cached, and are using predefined queue URLs,
+        # raise an error, don't try to create it.
+        if self.predefined_queue_urls:
+            raise UnknownQueueException
+
         # The SQS ListQueues method only returns 1000 queues.  When you have
         # so many queues, it's possible that the queue you are looking for is
         # not cached.  In this case, we could update the cache with the exact
         # queue name first.
-        if queue not in self._queue_cache:
-            self._update_queue_cache(queue)
+        self._update_queue_cache(queue)
         try:
             return self._queue_cache[queue]
         except KeyError:
```

Comment:
I would suggest to replace the modifications in this function with a simple additional condition in the `except` block:

```
if self.predefined_queue_urls:
    raise UnknownQueueException("Queue '{}' has not been declared in Predefined Queue URL mapping.".format(queue))
```